File: backend/src/main/java/com/example/informationstand/models/еxcursion/Excursion.java
<<<<<<< HEAD
package com.example.informationstand.models.еxcursion;

import jakarta.persistence.*;
import lombok.*;

import java.util.Date;

@Entity
@Table(name = "excursions")
@Getter
@Setter
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class Excursion {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String name;
    private String description;
    private String location;
    private int duration;
    private double cost;
    private Date startTime;
    private Date endTime;
    private double rating;
    private String reservedMembers;
    private int maxParticipants;
    @Column(name = "base64Image")
    private String base64Image;
    public Excursion(int maxParticipants) {
        this.maxParticipants = maxParticipants;
    }
}
=======
package com.example.informationstand.models.еxcursion;

import jakarta.persistence.*;
import lombok.*;

import java.util.Date;
import java.util.List;

@Entity
@Table(name = "excursions")
@Getter
@Setter
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class Excursion {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String name;
    private String description;
    private String location;
    private int duration;
    private double cost;
    private Date startTime;
    private Date endTime;
    private double rating;
    private String reservedMembers;
    private int maxParticipants;

}
>>>>>>> 36d864de4e886046ed1011ac5dadc54330919b12

File: flutter-app/lib/features/excursion/models/excursion.dart
import 'package:freezed_annotation/freezed_annotation.dart';
import 'package:information_stand/features/excursion/models/reserved_member.dart';

part 'excursion.freezed.dart';
part 'excursion.g.dart';

@freezed
abstract class Excursion with _$Excursion {
  const factory Excursion({
    required int id,
    required String name,
    required String description,
    required String location,
    required int duration,
    required double cost,
    required String startTime,
    required String endTime,
    required double rating,
    required List<ReservedMember> reservedMembers,
    required int maxParticipants,
    required String base64Image,
  }) = _Excursion;

  factory Excursion.fromJson(Map<String, dynamic> json) =>
      _$ExcursionFromJson(json);
}